pandas/core/internals.py

```diff
@@ -1406,7 +1405,7 @@ def to_native_types(self, slicer=None, na_rep='', float_format=None, decimal='.'
         if float_format and decimal != '.':
             formatter = lambda v : (float_format % v).replace('.',decimal,1)
         elif decimal != '.':
-            formatter = lambda v : ('%g' % v).replace('.',decimal,1)
+            formatter = lambda v : str(v).replace('.',decimal,1)
```

Comment:
no its fine. i doubt it was tested before. if this makes sense then go ahead and change (but document of course in whatsnew that was changed)